src/protocol/vm/state.rs

```diff
@@ -541,6 +544,111 @@ impl VMPoolState<PreCachedDB> {
 
         merged
     }
+
+    async fn get_amount_out(
+        &self,
+        sell_token: H160,
+        sell_amount: U256,
+        buy_token: H160,
+    ) -> Result<GetAmountOutResult, SimulationError> {
+        let mut sell_amount_respecting_limit = sell_amount;
+        let mut sell_amount_exceeds_limit = false;
+        let overwrites = self
+            .get_overwrites(
+                vec![sell_token, buy_token],
+                U256::from_big_endian(&(*MAX_BALANCE / rU256::from(100)).to_be_bytes::<32>()),
+            )
+            .await?;
+
+        let sell_amount_limit = self
+            .clone()
```

Comment:
Do we really need to clone here?

Reply:
true, now that we removed the mutable thingies I think we can remove it 👀